templates/terraform/custom_expand/bool_to_object.go.erb

```diff
@@ -0,0 +1,21 @@
+<%- # the license inside this block applies to this file
+	# Copyright 2020 Google Inc.
+	# Licensed under the Apache License, Version 2.0 (the "License");
+	# you may not use this file except in compliance with the License.
+	# You may obtain a copy of the License at
+	#
+	#     http://www.apache.org/licenses/LICENSE-2.0
+	#
+	# Unless required by applicable law or agreed to in writing, software
+	# distributed under the License is distributed on an "AS IS" BASIS,
+	# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	# See the License for the specific language governing permissions and
+	# limitations under the License.
+-%>
+func expand<%= prefix -%><%= titlelize_property(property) -%>(v interface{}, d TerraformResourceData, config *Config) (interface{}, error) {
+	if v == nil || !v.(bool) {
+        return nil, nil
+    }
+
+	return struct{}{}, nil
```

Comment:
We want to send an empty JSON object? Why would this be the case

Reply:
The API expects this
```
{
	"replication": {
		"automatic": {}
	}
}
```
But I didn't want that to be a nested object in Terraform, so I made `automatic` a boolean instead.  I don't think this is the only time we've done that.  I searched a bit and didn't find anything right away, but I know I've talked about it with someone once before, so I thought I'd make generic expand/flatten functions in case someone runs into it again.

Reply:
Sounds good. I just ran into this pattern in another API today, so it will be good to have